infra/modules/storage/access-control.tf

```diff
@@ -0,0 +1,34 @@
+# Block public access
+resource "aws_s3_bucket_public_access_block" "storage" {
+  bucket = aws_s3_bucket.storage.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+# Create policy for read/write access
+# Attach this policy to roles that need access to the bucket
+resource "aws_iam_policy" "storage_access" {
```

Comment:
Not super familiar in this area but [`HeadObject`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html) may be useful. My understanding is that's used by the SDK if you just want to check if a file exists.